### PR TITLE
[RFC] Create Threat Fieldset - Stage 3 Proposal

### PR DESCRIPTION
**Pull request by peasead**
- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)? Yes
- Have you followed the [contributor guidelines](https://github.com/elastic/ecs/blob/master/CONTRIBUTING.md)? Yes
- For proposing substantial changes or additions to the schema, have you reviewed the [RFC process](https://github.com/elastic/ecs/blob/master/rfcs/README.md)? Yes
- If submitting code/script changes, have you verified all tests pass locally using `make test`? Yes
- If submitting schema/fields updates, have you generated new artifacts by running `make` and committed those changes? Yes
- Is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed. Yes
- Have you added an entry to the [CHANGELOG.next.md](https://github.com/elastic/ecs/blob/master/CHANGELOG.next.md)? NA

Stage 3 criteria:

- [x] Opened pull request for this candidate revising the existing candidate
- [x] Included multiple real world example source documents
- [x] Existing or newly raised questions and concerns are addressed
- [x] No objections from sponsor, ECS team, or subject matter experts

[Preview of markdown proposal doc](https://github.com/peasead/ecs/blob/create-threat-stage-3/rfcs/text/0008-threat-intel.md)

### CODE REVIEW

**Review by ebeahan — COMMENTED** [review]
Now that more implementation work and testing has been done, do we have any resolution we can capture under [`Concerns`](https://github.com/peasead/ecs/blob/create-threat-stage-3/rfcs/text/0008-threat-intel.md#concerns)?

**Comment by peasead** [comment]
For  # 1 under `Concerns` WRT to malware name, can we use the [`threat.software.type`](https://www.elastic.co/guide/en/ecs/1.11/ecs-threat.html#field-threat-software-type) fieldset now?

@rylnd or @dcode under `Concerns`, is there anything that can be resolved WRT # 2?

**Comment by peasead** [comment]
> For # 1 under `Concerns` WRT to malware name and type, can we use the [`threat.software.type`](https://www.elastic.co/guide/en/ecs/1.11/ecs-threat.html#field-threat-software-type) and [`threat.software.name`](https://www.elastic.co/guide/en/ecs/1.11/ecs-threat.html#field-threat-software-name) fieldsets now?
> 
> @rylnd or @dcode under `Concerns`, is there anything that can be resolved WRT # 2?

Just pinging on this @rylnd / @dcode

File: rfcs/text/0008-threat-intel.md [hunk 1]
```diff
@@ -291,6 +310,8 @@ Stage 1: Identify potential concerns, implementation challenges, or complexity.
 -->
 
 1. How to best represent malware{name,family,type}. Current proposal is to use `threat.indicator.classification` to describe threat delivery (Hacktool etc.) and family name.
+  - This can be represented through the use of the [`threat.software.*`](https://www.elastic.co/guide/en/ecs/master/ecs-threat.html) fields.
+  - Awaiting [approval](https://github.com/elastic/ecs/pull/1480#issuecomment-889312434) of this recommendation.
 1. Field types (Ref: https://github.com/elastic/ecs/pull/1127#issuecomment-776126293)
   - `threatintel.indicator.*` (Filebeat module) will be normal field type and will be deprecated when nested field types are better supported in Kibana
   - `threat.indicator.*` (actual threat ECS fieldset) will be nested now and used for enriched doc
```
Comment (rylnd) [inline rfcs/text/0008-threat-intel.md:317]: @peasead I don't think this is still true; the plan for `threat.indicator` is to keep it as a normal field, as is already the case with `threatintel.indicator`. RE the migration below, I think the only change on the data shipper side is to  rename `threatintel.indicator` to `threat.indicator`.

`threat.enrichments` is the nested field that will cover the enrichment use case; the caveats about kibana support still apply there for Detection Alerts and users manually adopting the `threat.enrichments` fields, but it should not impact data shippers at this time.

Comment (peasead) [reply rfcs/text/0008-threat-intel.md:317]: Thanks, @rylnd so this isn't a "concern" any longer?

The concern may actually be that `threatintel.indicator.*` for the Threat Intel Filebeat module needs to be changed to `threat.indicator.*` once this is out of `[beta]`?

Comment (peasead) [reply rfcs/text/0008-threat-intel.md:317]: Roger.

I'll update the Concern.

@ebeahan FYI ☝️

**Comment by peasead** [comment]
@ebeahan we're ready for review

File: rfcs/text/0008-threat-intel.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
 # 0008: Cyber Threat Intelligence Fields
 <!-- Leave this ID at 0000. The ECS team will assign a unique, contiguous RFC number upon merging the initial stage of this RFC. -->
 
-- Stage: **2 (candidate)** <!-- Update to reflect target stage. See https://elastic.github.io/ecs/stages.html -->
-- Date: **2021-07-06** <!-- The ECS team sets this date at merge time. This is the date of the latest stage advancement. -->
+- Stage: **3 (candidate)** <!-- Update to reflect target stage. See https://elastic.github.io/ecs/stages.html -->
```
Comment (ebeahan) [inline rfcs/text/0008-threat-intel.md:4]:
```suggestion
- Stage: **3 (finished)** <!-- Update to reflect target stage. See https://elastic.github.io/ecs/stages.html -->
```

File: rfcs/text/0008-threat-intel.md [hunk 3]
```diff
@@ -291,14 +310,23 @@ Stage 1: Identify potential concerns, implementation challenges, or complexity.
 -->
 
 1. How to best represent malware{name,family,type}. Current proposal is to use `threat.indicator.classification` to describe threat delivery (Hacktool etc.) and family name.
+  - This can be represented through the use of the [`threat.software.*`](https://www.elastic.co/guide/en/ecs/master/ecs-threat.html) fields.
+  - Awaiting [approval](https://github.com/elastic/ecs/pull/1480#issuecomment-889312434) of this recommendation.
```
Comment (ebeahan) [inline rfcs/text/0008-threat-intel.md:314]: If we've come to an agreement, can we update or remove this line?

**Review by ebeahan — COMMENTED** [review]
I identified a couple of small items, but otherwise LGTM.

**Comment by ebeahan** [comment]
@devonakerr Would you provide your final review here as sponsor?

File: rfcs/text/0008-threat-intel.md (hunk 3 above)
Comment (peasead) [reply rfcs/text/0008-threat-intel.md]: Resolved.

**Review by ebeahan — APPROVED** [review]

**Comment by peasead** [comment]
We're ready for a final review.
@rylnd 
@devonakerr 
@dcode

**Review by devonakerr — APPROVED** [review]
Looks good, remarkable how far we've come with the schema in such a short period.

**Pull request merged**